### PR TITLE
RES-1521 password reset

### PR DESCRIPTION
**Pull request by edwh**
Lots of good stuff.

- Add Mediawiki image to the CircleCI configuration and enable wiki integration.  This means our tests now run with a real wiki.
- Change Mediawiki passwords to use the hashed Laravel password.
- Add a test of password recovery/reset.  This needs the code to be a bit more standard Laravel (avoid $_POST etc).
- Add a command to generate the Mediawiki commands to run from maintenance/ to migrate the passwords.
- Make the test users more unique, because we don't clean out the wiki users between tests.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -28,6 +28,26 @@ jobs:
       - image: mcr.microsoft.com/playwright:focal
         environment:
           NODE_ENV: development
+      - image: 'bitnami/mariadb:latest'
```
Comment (restart-tech) [inline .circleci/config.yml:31]: Any reason mariadb over mysql?  Fine by me - just thinking we're on mysql on prod, so...

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -490,52 +491,39 @@ public function reset()
...
+                $response['danger'] = __('passwords.match');
             } else {
-                $user = $User->where('recovery', '=', $recovery)->first();
-                if (! empty($user)) {
-                    $data = [
+                $email = $user->email;
+                $oldPassword = $user->password;
+
+                $update = $user->update([
                     'password' => crypt($pwd, '$1$'.strrev(md5(env('APP_KEY')))),
```
Comment (restart-tech) [inline app/Http/Controllers/UserController.php:509]: Does this need to change to use the Laravel Hash helper?  I thought we changed it already recently - perhaps not.

File: tests/Feature/Users/PasswordResetTest.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        $response->assertSeeText(__('passwords.sent'));
+
+        Notification::assertSentTo(
+            [$restarter], ResetPassword::class
+        );
+
+        $restarter->refresh();
+
+        // Invalid code
+        $response = $this->post('/user/reset', [
```
Comment (restart-tech) [inline tests/Feature/Users/PasswordResetTest.php:50]: There is no assertion on $response before it is given a new value on line 57 - maybe these lines are redundant?

File: tests/Feature/Users/PasswordResetTest.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        // Valid but mismatch passwords.
+        $response = $this->post('/user/reset', [
+            'recovery' => $restarter->recovery,
+            'password' => 'newpass',
+            'confirm_password' => 'mismatch'
+        ]);
+
+        $response->assertSeeText(__('passwords.match'));
+
+        // Valid - should redirect to login patch and dispatch password changed event to update the wiki.
```
Comment (restart-tech) [inline tests/Feature/Users/PasswordResetTest.php:74]: typo, patch = path?

**Review by restart-tech — COMMENTED** [review]
All looks good to me in general - should probably wait until having tested against real wiki before approving.

**Review by ngm — COMMENTED** [review]
All looks good, although one question around the use of the old password hashing method.

I think we should be updating it to Hash::make, like we were doing here? https://github.com/TheRestartProject/restarters.net/pull/388/files#diff-8e91484f56a38d906339650a2e3af6dbda84b6a43e6d889b792bd55b53a54e76L505

File: app/Http/Controllers/UserController.php (hunk 2 above)
Comment (edwh) [reply app/Http/Controllers/UserController.php:509]: Perhaps yes.  Fix pushed.

**Pull request merged**